### ISSUE TITLE
Ensure [Collection].Clear() actually deletes the records

### ISSUE THREAD

**Issue by sebastienros** [OrchardCMS/Orchard#5713]
For instance in `workflowDefinition.ActivityRecords.Clear()` the expectation is that the records are deleted. We need to confirm that they are not just removed from the collection.

**Comment by jtkech** [comment]
This is what I've experienced with workflows, child records are deleted. As far as I remember, this is what I've seen if [CascadeAllDeleteOrphan] is used on the collection, and as long as a child has no other parent. Idem if we delete the parent record (without first clearing the collection). For the last case, if the parent record isn't really deleted, as I've seen with somebody because it was a soft delete through a content item, then, of course here no child records are deleted

Just re-tried this with the last dev branch

- I created a workflow with 3 activities that I could see in the table with Ids: 1, 2, 3
- I re-edited the workflow and deleted the last activity
- Then, I could see in the table only 2 activities with Ids: 4, 5

For me, this means that, in the AdminController EditPost action, when you are doing this

    workflowDefinitionRecord.ActivityRecords.Clear();

The child records are really deleted, this before re-adding the only 2 remaining activities...

Best
